### PR TITLE
fix: fix content-type detection by doing a fall-back based on the ext…

### PR DESCRIPTION
**Pull request by satazor**
The JS IPFS gateway was only responding with the correct content-type for some mime-types, see https://github.com/sindresorhus/file-type#supported-file-types.
This commit now fall-backs to detecting based on the extension as well.

Note that SVGs aren't supported by the `file-type` module.

### CODE REVIEW

File: src/http/gateway/resources/gateway.js [hunk 1]
```diff
@@ -12,6 +12,21 @@ const Stream = require('readable-stream')
 const { resolver } = require('ipfs-http-response')
 const PathUtils = require('../utils/path')
 
+function detectContenType (ref, chunk) {
```
Comment (alanshaw) [inline src/http/gateway/resources/gateway.js]: Minor typo here `detectContenType` - missing "t" on "conten"

File: src/http/gateway/resources/gateway.js [hunk 2]
```diff
@@ -12,6 +12,21 @@ const Stream = require('readable-stream')
...
+  let fileSignature
+
+  // try to guess the filetype based on the first bytes
+  // note that `file-type` doesn't support svgs, thereore we assume it's a svg if ref looks like it
+  if (!ref.endsWith('.svg')) {
```
Comment (alanshaw) [inline src/http/gateway/resources/gateway.js]: Is this check necessary?

Comment (satazor) [reply src/http/gateway/resources/gateway.js]: Yes, file-type doesn’t support svgs and might actually return invalid mine types. They state that in their readme.

File: src/http/gateway/resources/gateway.js (hunk 1 above)
Comment (satazor) [reply src/http/gateway/resources/gateway.js]: Will fix once the workshop I’m attending on dweb ends. :p

File: src/http/gateway/resources/gateway.js (hunk 2 above)
Comment (alanshaw) [reply src/http/gateway/resources/gateway.js]: Ah ok, would you mind adding a comment to say that please?

Comment (alanshaw) [reply src/http/gateway/resources/gateway.js]: Just saw there’s already a comment there sorry!

**Comment by satazor** [comment]
The typo should be fixed!

**Pull request merged**